src/services/gitlab/gitlabClient/GitLabClient.ts

```diff
@@ -0,0 +1,31 @@
+import axios from 'axios';
+import { injectable } from 'inversify';
+
+@injectable()
+export class GitLabClient {
+  protected headers: { [header: string]: string };
+  protected host: string;
+  protected timeout: number;
+
+  constructor({ token, host = 'https://gitlab.com', timeout = 20000 }: ClientOptions) {
```

Comment:
20 seconds as a timeout is quite high. I'd set it to 5 seconds to not block a user if something happens.